charts/datadog/templates/_containers-common-env.yaml

```diff
@@ -17,6 +17,10 @@
 - name: DD_KUBERNETES_KUBELET_HOST
 {{ toYaml .Values.datadog.kubelet.host | indent 2 }}
 {{- end }}
+{{- if .Values.datadog.kubelet.core_check }}
+- name: DD_KUBERNETES_KUBELET_CORE_CHECK_ENABLED
+  value: {{ .Values.datadog.kubelet.core_check | quote }}
+{{- end }}
```

Comment:
Since this should only have baring on the core agent, should this instead be moved to https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/_container-agent.yaml#L58